### PR TITLE
Index scan ops

### PR DESCRIPTION
**Pull request by jeffreylovitz**

### CODE REVIEW

File: src/execution_plan/ops/op_index_scan.c [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  return OP_OK;
+}
+
+OpResult IndexScanReset(OpBase *ctx) {
+  IndexScan *indexScan = (IndexScan*)ctx;
+
+  /* Restore original node. */
+  *indexScan->node = indexScan->_node;
+  free(indexScan->iter);
```
Comment (swilly22) [inline src/execution_plan/ops/op_index_scan.c]: Use iterator reset.

File: src/execution_plan/ops/op_index_scan.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  indexScan->iter = skiplistIterateAll(indexScan->index->sl);
+
+  return OP_OK;
+}
+
+void IndexScanFree(OpBase *op) {
+  IndexScan *indexScan = (IndexScan *)op;
+  free(indexScan->iter);
```
Comment (swilly22) [inline src/execution_plan/ops/op_index_scan.c]: Use iterator free

File: src/execution_plan/execution_plan.c [hunk 3]
```diff
@@ -327,10 +327,13 @@ ExecutionPlan *NewExecutionPlan(RedisModuleCtx *ctx, const char *graph_name, AST
     if(ast->matchNode) BuildGraph(graph, ast->matchNode->graphEntities);
 
     ExecutionPlan *execution_plan = (ExecutionPlan*)calloc(1, sizeof(ExecutionPlan));
-    
-    /* List of operations. */    
-    FT_FilterNode *filterTree = NULL;
 
+    // Build the filter tree if necessary first, as it will be used to find applicable indices
+    if(ast->whereNode != NULL) {
```
Comment (swilly22) [inline src/execution_plan/execution_plan.c]: Drop changes to the execution plan,
I'll take care of that part.

**Comment by jeffreylovitz** [comment]
@swilly22 I think that the code changes in this PR are complete, but it has dependencies on other code that currently only exists in pull requests #105 and #106.

As such, this branch will fail to compile until those PRs are also merged or the relevant code from them has been cherry-picked to here (which I can do, if you'd like to keep the merges fairly modular!).

**Comment by jeffreylovitz** [comment]
This PR has been rolled into PR #106

**Pull request closed without merging**